pyee/_base.py

```diff
@@ -118,7 +123,13 @@ def once(self, event, f=None):
         """
         def _wrapper(f):
             def g(*args, **kwargs):
-                self.remove_listener(event, f)
+                with self._lock:
+                    # Check that the event wasn't removed already right
+                    # before the lock
+                    if event in self._events and f in self._events[event]:
+                        self._remove_listener(event, f)
+                    else:
+                        return None
```

Comment:
Yeah OK I see what you mean here. Interesting, I wouldn't have expected the edge case we saw given this ordering (though in a threaded context without locks it makes sense). Guess I should have looked more closely!

Mulling this over, I *think* shrugging and returning None is OK, because if the edge case we're talking about triggers it probably means that the handler has either already been called or was  intentionally removed between the function executing and the lock yielding.